react-app/src/components/SettingsModal/index.js

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import ActionIcon from 'components/ActionIcon';
+import BaseModal from 'components/BaseModal';
+import Logo from 'components/Logo';
+import RadioButton from 'components/RadioButton';
+import SettingsOptionLabel from 'components/SettingsOptionLabel';
+
+import styles from './index.module.scss';
+
+export default class SettingsModal extends React.PureComponent {
+  static propTypes = {
+    /**
+     * The username to display
+     */
+    username: PropTypes.string.isRequired,
+
+    /**
+     * The function to call when hide icon is tapped
+     */
+    onHide: PropTypes.func.isRequired,
+
+    /**
+     * The function to call when logout option is tapped
+     */
+    onSignOut: PropTypes.func.isRequired,
+  }
+
+  state = {
+    visible: false,
+    daily: true,
+  };
+
+  _setDailyNotifications = () => this.setState({ daily: true });
+  _setWeeklyNotifications = () => this.setState({ daily: false });
```

Comment:
I think this is fine for now because we only have two options, but it's not intuitive at first sight that `daily: false` means `weekly: true`. I'd use another variable instead that can be set to a couple of constants called `DAILY` and `WEEKLY`. Like this:
```javascript
_setDailyNotifications = () => this.setState({ notifications: DAILY });
_setWeeklyNotifications = () => this.setState({ notifications: WEEKLY });
```

Thoughts?

Reply:
You're right. I think that maybe, it will be OK to add `DAILY` and `WEEKLY` constants inside the notifications module (which it is not created yet) and import the constants here. I will need to chage so the PR #44 to use this constants, is this OK?